lms/djangoapps/ccx/views.py

```diff
@@ -196,42 +198,51 @@ def save_ccx(request, course, ccx=None):
     if not ccx:
         raise Http404
 
-    def override_fields(parent, data, graded, earliest=None):
+    def override_fields(parent, data, graded, earliest=None, **kwargs):
         """
         Recursively apply CCX schedule data to CCX by overriding the
         `visible_to_staff_only`, `start` and `due` fields for units in the
         course.
         """
+        delete_ccx_ids = kwargs.get('delete_ccx_ids', [])
+
         blocks = {
             str(child.location): child
             for child in parent.get_children()}
+
         for unit in data:
             block = blocks[unit['location']]
             override_field_for_ccx(
                 ccx, block, 'visible_to_staff_only', unit['hidden'])
+
             start = parse_date(unit['start'])
             if start:
                 if not earliest or start < earliest:
                     earliest = start
                 override_field_for_ccx(ccx, block, 'start', start)
             else:
-                clear_override_for_ccx(ccx, block, 'start')
+                delete_ccx_ids.append(get_override_for_ccx(ccx, block, 'start_id'))
+                clear_ccx_field_info_from_ccx_map(ccx, block, 'start')
+
             due = parse_date(unit['due'])
             if due:
                 override_field_for_ccx(ccx, block, 'due', due)
             else:
-                clear_override_for_ccx(ccx, block, 'due')
+                delete_ccx_ids.append(get_override_for_ccx(ccx, block, 'due_id'))
+                clear_ccx_field_info_from_ccx_map(ccx, block, 'due')
 
             if not unit['hidden'] and block.graded:
                 graded[block.format] = graded.get(block.format, 0) + 1
 
             children = unit.get('children', None)
             if children:
-                override_fields(block, children, graded, earliest)
-        return earliest
+                override_fields(block, children, graded, earliest, **kwargs)
+        return earliest, kwargs
```

Comment:
Instead of returning kwargs, this should just return the ccx ids we intend to delete